### PR TITLE
Refactoring/advisory board decision remove state objects

### PR DESCRIPTION
**Pull request by paullocknimble**
Removed last remaining state objects so the domain is mapped to the database through ef config.

### CODE REVIEW

File: Dfe.Academies.Academisation.Data/AcademisationContext.cs [hunk 1]
```diff
@@ -44,28 +46,28 @@ public override int SaveChanges()
 		return base.SaveChanges();
 	}
 
-	public EntityEntry<T> ReplaceTracked<T>(T baseEntity) where T : BaseEntity
-	{
-		var entity = ChangeTracker
-			.Entries<T>()
-			.SingleOrDefault(s => s.Entity.Id == baseEntity.Id);
+	//public EntityEntry<T> ReplaceTracked<T>(T baseEntity) where T : BaseEntity
```
Comment (dneed-nimble) [inline Dfe.Academies.Academisation.Data/AcademisationContext.cs:49]: Shall we remove this?

File: Dfe.Academies.Academisation.Data/AcademisationContext.cs [hunk 2]
```diff
@@ -88,15 +90,39 @@ private void SetModifiedAndCreatedDates()
 	{
 		var timestamp = DateTime.UtcNow;
 
-		var entities = ChangeTracker.Entries<BaseEntity>().ToList();
+		var entities2 = ChangeTracker.Entries<AdvisoryBoardDeferredReasonDetails>().ToList();
```
Comment (dneed-nimble) [inline Dfe.Academies.Academisation.Data/AcademisationContext.cs:93]: Would be good to talk through this to understand it

File: Dfe.Academies.Academisation.Data/Repositories/AdvisoryBoardDecisionRepository.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+		public async Task<ConversionAdvisoryBoardDecision?> GetTransferProjectDecsion(int projectId)
+		{
+			try
+			{
+return await DefaultIncludes().SingleOrDefaultAsync(x => x.AdvisoryBoardDecisionDetails.TransferProjectId == projectId);
+			}
+			catch (Exception ex)
+			{
+				var e = ex;
+				throw;
```
Comment (dneed-nimble) [inline Dfe.Academies.Academisation.Data/Repositories/AdvisoryBoardDecisionRepository.cs:48]: Shall we throw something specific

File: Dfe.Academies.Academisation.Data/AcademisationContext.cs (hunk 2 above)
Comment (dneed-nimble) [reply Dfe.Academies.Academisation.Data/AcademisationContext.cs:93]: Maybe rename to DeclinedEntities etc

**Review by elielijah321 — APPROVED** [review]

File: Dfe.Academies.Academisation.Data/Repositories/AdvisoryBoardDecisionRepository.cs (hunk 3 above)
Comment (paullocknimble) [reply Dfe.Academies.Academisation.Data/Repositories/AdvisoryBoardDecisionRepository.cs]: removed

File: Dfe.Academies.Academisation.Data/AcademisationContext.cs (hunk 2 above)
Comment (paullocknimble) [reply Dfe.Academies.Academisation.Data/AcademisationContext.cs]: renamed with comment to explain why they are there

File: Dfe.Academies.Academisation.Data/AcademisationContext.cs (hunk 1 above)
Comment (paullocknimble) [reply Dfe.Academies.Academisation.Data/AcademisationContext.cs]: removed

**Pull request merged**